File: [프로그래머스] 2018 KAKAO BLIND RECRUITMENT_[1차] 뉴스 클러스터링.py
def solution(str1, str2):
    str1Arr = [str1[i:i+2].upper() for i in range(len(str1)-1) if str1[i:i+2].isalpha()]
    str2Arr = [str2[i:i+2].upper() for i in range(len(str2)-1) if str2[i:i+2].isalpha()]
    str1Dic = {i:str1Arr.count(i) for i in str1Arr}
    str2Dic = {i:str2Arr.count(i) for i in str2Arr}
    kyo, hap = 0, 0
    for i in str1Dic.keys():
        if i in str2Dic.keys():
            kyo += min(str1Dic[i], str2Dic[i])
            hap += max(str1Dic[i], str2Dic[i])
        if i not in str2Dic.keys():
            hap += str1Dic[i]
    for j in str2Dic.keys():
        if j not in str1Dic.keys():
            hap += str2Dic[j]
    return 65536 if hap == 0 else int(65536 * (kyo / hap))

File: [프로그래머스] 2018 KAKAO BLIND RECRUITMENT_[1차] 뉴스 클러스터링.js
function solution(str1, str2) {
    // 알파벳 체크 함수
    function isalpha(c){
        return (((c >= 'a') && (c <= 'z')) || ((c >= 'A') && (c <= 'Z')));
    }
    
    // 알파벳을 체크해 알파벳으로만 이루어진 쌍을 만들고 배열에 담아주는 함수
    function checkAlpha(string){
        let arr = []
        for(let i = 0; i < string.length - 1; i++){
            if(isalpha(string[i]) && isalpha(string[i+1])){
                arr.push(string[i].toUpperCase() + string[i+1].toUpperCase());
            }
        }
        return arr;
    }
   
    let arr1 = checkAlpha(str1);
    let arr2 = checkAlpha(str2);
    let setArr1 = [...new Set(arr1)];
    let setArr2 = [...new Set(arr2)];
    let kyo = 0;
    let hap = 0;
    
    for(let i of setArr1){
        if(arr2.includes(i) === true){
            kyo += Math.min(arr1.filter(el => el === i).length, arr2.filter(el => el === i).length);
            hap += Math.max(arr1.filter(el => el === i).length, arr2.filter(el => el === i).length);
        } else {
            hap += arr1.filter(el => el === i).length;
        }
    }
    for(let j of setArr2){
        if(arr1.includes(j) === false){
            hap += arr2.filter(el => el === j).length;
        }
    }    
    return hap === 0 ? 65536 : parseInt(kyo/hap * 65536);
}